libc/CMakeLists.txt

```diff
@@ -706,33 +703,18 @@ if (USE_DEBUG_MALLOC)
 endif ()
 
 # This project (and its dependents) require the enclave headers and
-# library.
-target_link_libraries(oelibc PUBLIC oecore)
-
-# As a pre-build step, place the musl LIBC headers.
-add_dependencies(oelibc oelibc_includes)
+# library, and the musl headers.
+target_link_libraries(oelibc
+  PUBLIC oecore oelibc_includes
+  PRIVATE oelibasm)
 
 target_include_directories(oelibc PRIVATE
+  internal
   ${PROJECT_BINARY_DIR}/3rdparty/musl/musl/src/internal
   ${PROJECT_BINARY_DIR}/3rdparty/musl/musl/arch/x86_64)
 
-# This project and its dependents require include/libc in the include path.
-#
-# Note the difference between build time (absolute in build tree) and
-# install time (relative to install prefix).
-target_include_directories(oelibc PUBLIC
-  $<BUILD_INTERFACE:${OE_INCDIR}/openenclave/libc>
-  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/openenclave/3rdparty/libc>)
-
-# When building this project, we also need a number of internal includes:
-target_include_directories(oelibc PRIVATE
-  ${CMAKE_CURRENT_SOURCE_DIR}/internal
-  ${PROJECT_BINARY_DIR}/3rdparty/musl/musl/src/internal)
-
-# Assemble lib-archive into proper directory.
 set_property(TARGET oelibc PROPERTY
   ARCHIVE_OUTPUT_DIRECTORY ${OE_LIBDIR}/openenclave/enclave)
 
-# Install lib-archive upon install-time.
-install(TARGETS oelibc EXPORT openenclave-targets
+install(TARGETS oelibc oelibasm EXPORT openenclave-targets
```

Comment:
Does this mean there is going to be an 'oelibasm.a' in the installation?
If so, we need to update pkgconfig definitions.

I would prefer that we don't introduce new libraries; we are leaning in the other direction of combining libraries (e.g. we want to combine the 3 mbedtls oe libraries into 1)

Reply:
Explicitly no, as `oelibasm.a` does not exist since `oelibasm` is an `OBJECT` library, not a `STATIC` library. At some point I had to add it to the list of exported targets because CMake complained, but is on my list of TODOs to verify the end result of all our export statements (and that may actually be a separate issue, as we know they're already incorrect).

Reply:
Also, we should be automatically generating the `pkg-config` file instead of maintaining it manually 😄

Reply:
I've added generating/checking `pkg-config` to #887.